ftp/src/test/scala/akka/stream/alpakka/ftp/BaseFtpsSpec.scala

```diff
@@ -29,7 +30,7 @@ trait BaseFtpsSpec extends FtpsSupportImpl with BaseSpec {
                                     emitTraversedDirectories: Boolean): Source[FtpFile, NotUsed] =
     Ftps.ls(basePath, settings, branchSelector, emitTraversedDirectories)
 
-  protected def retrieveFromPath(path: String): Source[ByteString, Future[IOResult]] =
+  protected def retrieveFromPath(path: String, fromRoot: Boolean = false): Source[ByteString, Future[IOResult]] =
```

Comment:
I didn't see any use of the `fromRoot` parameter. Could it be dropped?

Reply:
It is used [here](https://github.com/akka/alpakka/pull/1668/files/ae5e67eadc9e710ef10e46154fba71373d5ce760#diff-95180c276c56d6b228c82b00d495d22eR185).

This is related to the fact that we can't write to the root folder of the SFTP server. In that test the path comes from ftpFiles and starts with `upload/` already (in the case of SFTP specs). So `upload/` [shouldn't be added again](https://github.com/akka/alpakka/pull/1668/files/ae5e67eadc9e710ef10e46154fba71373d5ce760#diff-9da90ed2137d0a1874b20e40e6b8d2a5R33) by `retrieveFromPath`. I tried to make things transparent from the test implementer perspective but I couldn't find a better solution than to leak an implementation detail here. You might have a better idea though.

This is also why [testing the path inside an FtpFile](https://github.com/akka/alpakka/pull/1668/files/ae5e67eadc9e710ef10e46154fba71373d5ce760#diff-95180c276c56d6b228c82b00d495d22eR145) is tricky since it may contain an `upload/` part or not depending on whether we are running an FTP test or an SFTP one.

Reply:
OK.